Assets/AppCenter/Editor/MsalDependency.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using UnityEditor;
+using UnityEngine;
+
+/// <summary>
+/// This file is used to define dependencies, and pass them along to a system which can resolve dependencies.
+/// </summary>
+public class MsalDependency
+{
+    private const string MsalVersion = "0.3.1-alpha";
+    private const string SupportLibVersion = "27.1.1";
+    private const string GsonVersion = "2.8.4";
+    private const string IdentityCommonVersion = "0.0.10-alpha";
+    private const string NimbusVersion = "5.7";
+
+    static void SetupDependencies()
+    {
+#if UNITY_ANDROID
+        // Setup the resolver using reflection as the module may not be available at compile time.
+        Type versionHandler = Type.GetType("Google.VersionHandler, Google.VersionHandler, Version=1.2.0.0, Culture=neutral, PublicKeyToken=null");
```

Comment:
Since this logic is now in two places, let's move it to some common place. Off the top of my head, we can either
- pass `playServicesSupport` instance to `SetupDependencies`
- or make this an abstract class with a base method `InitializePlayServices`

Reply:
So for the first option, we could initialize the `playServicesSupport` instance in AppCenterPreBuild.cs, and pass it to `SetUpAuth` and `SetUpPush`?

Reply:
The refactoring will be coming later. Will capture it as a task on the board.